rust/private/rustc.bzl

```diff
@@ -1823,7 +1839,7 @@ def _add_native_link_flags(args, dep_info, linkstamp_outs, ambiguous_libs, crate
     use_pic = _should_use_pic(cc_toolchain, feature_configuration, crate_type, compilation_mode)
 
     if toolchain.os == "windows":
-        make_link_flags = _make_link_flags_windows
+        make_link_flags = _make_link_flags_windows_msvc if toolchain.target_triple.abi == "msvc" else _make_link_flags_windows_gnu
```

Comment:
Does this need to match the `target_triple` or `exec_triple`? `toolchain.os`, which should be deleted, represents the `exec_triple`.

Reply:
`target_triple`, see above comment.  I did not realize that `toolchain.os` represented `exec_triple`.  My setup that lead me to making this PR, I declared `os` on my custom toolchain based on target, so evidently my "fix" was wrong to start with 😅

Reply:
This actually turned out to not be entirely sure. I just merged the other PR which should make the intent clearer